src/selectors/composedSelectors.js

```diff
@@ -0,0 +1,218 @@
+import Immutable from 'immutable';
+import { createSelector } from 'reselect';
+import _ from 'lodash';
+import MAX_SAFE_INTEGER from 'max-safe-integer'
+
+export const hasPreviousSelector = {
+  creator: ({currentPageSelector}) => {
```

Comment:
This may be too clever, but I wonder if this would be easier to write/read by attaching `dependencies` directly to the `creator` function:

```javascript
export const hasPreviousSelector = ({ currentPageSelector }) =>
  createSelector(
    currentPageSelector,
    currentPage => currentPage > 1);
hasPreviousSelector.dependencies = ["currentPageSelector"];
```